### PR TITLE
Use addressable-resolver aggregated clusterrole

### PR DESCRIPTION
**Pull request by Abd4llA**
Fixes knative/eventing#916

## Proposed Changes

* Add addressable-resolver role to controller-manager serviceaccount
* Add addressable-resolver role to gcppubsub-controller-manager serviceaccount

**Release Note**

```release-note
Add addressable-resolver cluster role for source controllers service accounts 
```

### CODE REVIEW

File: config/203-resolver-clusterrolebinding.yaml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+metadata:
+  name: eventing-sources-addressable-resolver
+subjects:
+- kind: ServiceAccount
+  name: controller-manager
+  namespace: knative-sources
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: addressable-resolver
```
Comment (Harwayne) [inline config/203-resolver-clusterrolebinding.yaml]: New line.

File: contrib/gcppubsub/config/201-clusterrole.yaml [hunk 2]
```diff
@@ -55,13 +55,6 @@ rules:
...
   - watch
 
-- apiGroups:
-  - eventing.knative.dev
-  resources:
-  - channels
-  - brokers
-  verbs: *readOnly
-
 - apiGroups:
```
Comment (Harwayne) [inline contrib/gcppubsub/config/201-clusterrole.yaml]: These can be removed too.

File: contrib/gcppubsub/config/203-resolver-clusterrolebinding.yaml [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: eventing-sources-gcppubsub-resolver
+subjects:
+- kind: ServiceAccount
+  name: gcppubsub-controller-manager
+  namespace: knative-sources
+roleRef:
```
Comment (Harwayne) [inline contrib/gcppubsub/config/203-resolver-clusterrolebinding.yaml]: Add a comment that this is defined in knative/eventing (perhaps include a link?).

File: config/203-resolver-clusterrolebinding.yaml [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: eventing-sources-addressable-resolver
+subjects:
+- kind: ServiceAccount
+  name: controller-manager
+  namespace: knative-sources
+roleRef:
```
Comment (Harwayne) [inline config/203-resolver-clusterrolebinding.yaml]: Add a comment that this is defined in knative/eventing (perhaps include a link?).

**Comment by Abd4llA** [comment]
@adamharwayne 
* I'm going to combine both clusterrolebinding objects in a single `-clusterrolebinding.yaml` file.
* I didn't change the camel or kafka sources, only gcppubsub. I suggest we create follow up issues for those later so that they get thoroughly tested each on their own?

**Comment by Harwayne** [comment]
> * I'm going to combine both clusterrolebinding objects in a single `-clusterrolebinding.yaml` file.
> * I didn't change the camel or kafka sources, only gcppubsub. I suggest we create follow up issues for those later so that they get thoroughly tested each on their own?

SGTM.

Also, please use @Harwayne, instead of adamharwayne.

**Comment by Abd4llA** [comment]
Oops! Sorry :D

File: contrib/gcppubsub/config/202-clusterrolebinding.yaml [hunk 5]
```diff
@@ -24,3 +24,20 @@ roleRef:
...
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: eventing-sources-gcppubsub-resolver
+subjects:
+- kind: ServiceAccount
+  name: gcppubsub-controller-manager
+  namespace: knative-sources
+# An aggregated ClusterRole for all Addressable CRDs. 
+# Ref: https://github.com/knative/eventing/blob/master/config/200-addressable-resolvers-clusterrole.yaml
```
Comment (n3wscott) [inline contrib/gcppubsub/config/202-clusterrolebinding.yaml]: SO COOL, thanks for doing this work.

**Pull request merged**